src/assets/scripts/client/aircraft/Pilot/Pilot.js

```diff
@@ -461,24 +457,10 @@ export default class Pilot {
             nextAltitude = this._fms.flightPlanAltitude;
         }
 
-        if (nextAltitude === INVALID_NUMBER) {
-            const readback = {};
-            readback.log = 'unable to climb via SID, no altitude assigned';
-            readback.say = 'unable to climb via SID, no altitude assigned';
-
-            return [false, readback];
-        }
-
-        if (typeof nextAltitude !== 'number') {
-            return [false, `unable to climb to altitude of ${nextAltitude}`];
-        }
+        const response = aircraftModel.validateNextAltitude(nextAltitude);
 
-        if (!aircraftModel.model.isAbleToMaintainAltitude(nextAltitude)) {
-            const readback = {};
-            readback.log = `unable to maintain ${nextAltitude} due to performance`;
-            readback.say = `unable to maintain ${radio_altitude(nextAltitude)} due to performance`;
-
-            return [false, readback];
+        if (response && !response[0]) {
```

Comment:
same principle here, if `.validateNextAltitude()` always returns an array this if is much simpler.

Reply:
done